packit_service/worker/helpers/open_scan_hub.py

```diff
@@ -78,42 +80,41 @@ def handle_scan(self):
 
             build_dashboard_url = get_copr_build_info_url(self.build.id)
 
-            output = self.copr_build_helper.api.run_osh_build(
-                srpm_path=paths[1],
-                base_srpm=paths[0],
-                comment=f"Submitted via Packit Service for {build_dashboard_url}",
-            )
-
-            if not output:
-                logger.debug("Something went wrong, skipping the reporting.")
-                return
-
-            logger.info("Scan submitted successfully.")
-
-            response_dict = self.parse_dict_from_output(output)
-
-            logger.debug(f"Parsed dict from output: {response_dict} ")
-
-            scan = None
-            if id := response_dict.get("id"):
-                if self.build.scan:
-                    # see comment https://github.com/packit/packit-service/issues/2604#issuecomment-2444321483
-                    logger.debug(
-                        f"Scan for build {self.build.id} already submitted, "
-                        "scan task_id {self.build.scan.task_id}."
+            try:
+                with self.build.add_scan_transaction() as scan:
+                    output = self.copr_build_helper.api.run_osh_build(
+                        srpm_path=paths[1],
+                        base_srpm=paths[0],
+                        comment=f"Submitted via Packit Service for {build_dashboard_url}",
                     )
-                    return
-                scan = self.build.add_scan(task_id=id)
-            else:
-                logger.debug(
-                    "It was not possible to get the Open Scan Hub task_id from the response.",
-                )
 
-            if not (url := response_dict.get("url")):
-                logger.debug("It was not possible to get the URL from the response.")
-                return
-            if url and scan:
-                scan.set_url(url)
+                    if not output:
+                        logger.debug("Something went wrong, skipping the reporting.")
+                        return
+
+                    logger.info("Scan submitted successfully.")
+
+                    response_dict = self.parse_dict_from_output(output)
+
+                    logger.debug(f"Parsed dict from output: {response_dict} ")
+
+                    scan = None
```

Comment:
What is this supposed to achieve?

Reply:
You mean try/catching the `IntegrityException`?

Reply:
I mean why are you assigning `None` to target variable of the context. But I don't understand the `IntegrityException` either.